### ISSUE TITLE
No results in evaluation P R and mAP are 0

### ISSUE THREAD

**Issue by qazi876** [tanluren/yolov3-channel-and-layer-pruning#84]
Hello @zbyuan Please check my am i getting the P R and mAP zero. Below is the values after each epoch 

     Epoch   gpu_mem      GIoU       obj       cls     total      soft    rratio   targets  img_size
  0%|                                                                                         | 0/1355 [00:00<?, ?it/s]learning rate: 1e-06
       0/2     7.65G      2.44      1.74      41.9      46.1         0         0         3       416: 100%|█| 1355/1355
Reading image shapes: 100%|███████████████████████████████████████████████████████| 5417/5417 [00:17<00:00, 303.63it/s]
               Class    Images   Targets         P         R       mAP        F1: 100%|█| 339/339 [01:13<00:00,  4.63it
                 all  5.42e+03  6.14e+03         0         0         0         0

**Comment by zbyuan** [comment]
Load pre-training weights？Date problem？QQ group : 734912150

**Comment by qazi876** [comment]
> Load pre-training weights？Date problem？QQ group : 734912150

I have this error when The code is evaluating after every epoch

**Comment by zbyuan** [comment]
Um, you can train yourself with dozens of data .  Encountered the same problem？May be a data problem

**Comment by tanluren** [comment]
check the train_batch0.jpg and test_batch0.jpg to see if you prepair the data right

**Comment by suzhr3** [comment]
@zbyuan @tanluren Hello, thanks for your work!
I have this problem, too，and I also check the train_batch0.jpg and test_batch0.jpg, and both of them are prepaired correctly. I also try to train with yolov3.weight and train with no pre-trained model, the results are the same， R R and mAP are 0. In addition, when I train the network, a hint occur: 
    "Gradient overflow.  Skipping step, loss scaler 0 reducing loss scale to 32768.0".
I don't know if this hint has anything to do with the mistake above.

**Comment by qazi876** [comment]
> check the train_batch0.jpg and test_batch0.jpg to see if you prepair the data right
Hi @zbyuan @tanluren 
Is it possible that I talk to you on wechat Please. I can explain my problem. 

Actually I want to train the teacher model on different classes and student model on different classes and in testing I wan to predict both.

**Comment by zhuhuangru** [comment]
Please check your dataset. These are some difference between officail YOLO dataset and ultralytics version Dataset.

**Comment by zhangnan-hust** [comment]
Would you please tell me how different between the official yolo dataset and ultralytics version dataset, and how to generate the dataset and labels?
I met the same problem, and  I used the voc_label.py to create the trainval.txt, test.txt which including the path of images and labels which including the classes and the information of bbox.

**Comment by wenhaotang** [comment]
> Load pre-training weights？Date problem？QQ group : 734912150

Hi @zbyuan , the QQ group : 734912150 is not found now. How can I find the group? : )

**Comment by zbyuan** [comment]
@wenhaotang Your qq number

**Comment by wenhaotang** [comment]
> @wenhaotang Your qq number

@zbyuan  My qq number: 423184965
Thanks!
